haskell/cabal.bzl

```
@@ -256,26 +255,40 @@ def _prepare_cabal_inputs(
     #   the linker used by `hsc2hs` (which will be our own wrapper script which
     #   eventually calls `gcc`, etc.).
     if hs.toolchain.static_runtime:
-        args.add("--ghc-option=-fPIC")
+        extra_args.append("--ghc-option=-fPIC")
 
         if not hs.toolchain.is_windows:
-            args.add("--ghc-option=-fexternal-dynamic-refs")
+            extra_args.append("--ghc-option=-fexternal-dynamic-refs")
 
     if hs.toolchain.fully_static_link:
-        args.add("--hsc2hs-option=--lflag=-static")
+        extra_args.append("--hsc2hs-option=--lflag=-static")
 
     if hs.features.fully_static_link:
-        args.add("--ghc-option=-optl-static")
+        extra_args.append("--ghc-option=-optl-static")
 
-    args.add("--")
-    args.add_all(package_databases, map_each = _dirname, format_each = "--package-db=%s")
-    args.add_all(direct_include_dirs, format_each = "--extra-include-dirs=%s")
-    args.add_all(direct_lib_dirs, format_each = "--extra-lib-dirs=%s", uniquify = True)
+    path_args = [
+        "--package-db=" + _dirname(db)
+        for db in package_databases.to_list()
+    ] + [
+        "--extra-include-dirs=" + d
+        for d in direct_include_dirs.to_list()
+    ] + _uniquify(["--extra-lib-dirs=" + d for d in direct_lib_dirs])
     if with_profiling:
-        args.add("--enable-profiling")
+        path_args.append("--enable-profiling")
```

```suggestion
        extra_args.append("--enable-profiling")
```
This doesn't contain a path, so doesn't need the execroot prepended.